### DISCUSSION TITLE
Deluge Category Label Limitation

### DISCUSSION THREAD

**Discussion by neo-neo1** [pymedusa/Medusa#9842] [Ideas & Feature requests]
When using Deluge as the torrent client the labeling options Medusa provides is extremely limiting.

Medusa instructs to enable the label plugin which is quite basic, unmaintained, and limiting. This plugin is basically abandoned as everyone has switched to the LabelPlus plugin instead. It would be very useful if Medusa would support the LabelPlus plugin as well.

[Screenshot of Label field in reffrence](https://i.imgur.com/ecnaS3l.png)

At the very least please support either or Plugin (Label or LabelsPlus). Current users of LabelPlus are forced to downgrade and loose vital features just to have Medusa support.

**Comment by OmgImAlexis** [comment]
> quite basic, unmaintained, and limiting

I've been using it for years and don't have any issues. What exactly isn't supported?

**Reply by neo-neo1** [reply]
> > quite basic, unmaintained, and limiting
> 
> I've been using it for years and don't have any issues. What exactly isn't supported?

And when you set it up years ago, LabelPlus likely didn't exist or was in it's infancy. 
Comparing the numerous added options and improvements which LabelPlus offers over Labels is self evident and the benefits are obvious. 

If you're configuring Deluge with Medusa **today** you're forced to use the inferior Label plugin instead of the current standard, LabelPlus. Why not support both? For current users of LabelPlus (for reasons outside of Medusa) you're forced to downgrade and loose many vital features we rely on just to have Medusa support. Furthermore, when I looked into this the Label plugin was mostly unmaintained and LabelPlus was touted as the replacement.  At the very least please add support for either plugin so the users can freely choose and not be shoehorned into inferiority. 

A user of Label plugin looses absolutely nothing switching to LabelPlus. Yet, a user of LabelPlus would break numerous usability options being forced to downgrade to Labels. My entire Deluge setup revolves around LabelPlus, it's absolutely crucial.

**Reply by p0psicles** [reply]
I tried installing the labelplus plugin in a docker container with deluge 2. But no success.
Also does labelplus have support for the webui in v2?

**Reply by neo-neo1** [reply]
> I tried installing the labelplus plugin in a docker container with deluge 2. But no success.
> Also does labelplus have support for the webui in v2?

The v2 fork [is here](https://github.com/bdutro/deluge-labelplus/releases)
I'm currently using it on linuxserver/deluge container without issue 

You can rename the .egg filename to match specific python3 version of your system if needed.

**Reply by p0psicles** [reply]
It didn't want to install through the webui upload. So I dropped it manuall in the plugins folder.
But I don't know what to look for now? SHould it display in the webui's plugins list?
Should there be new options available?

**Reply by neo-neo1** [reply]
> didn't want to install through the webui upload. So I dropped it manuall in the plugins folder.
> But I don't know what to look for now? SHould it display in the webui's plugins list?
> Should there be new options available?

I believe LabelsPlus does not work in WebUI. 
You have to connect to your Docker Deluge with an operating system based client.

[Photo of what you should look for](https://i.imgur.com/CcHycwq.png)

When using an OS based client you have to also place the LabelPlus .egg file into the respective plugin folder of your client installation.
Then ensure LabelPlus is enabled under "Preferences" -> "Plugins" 
Then configure the plugin from the LabelPlus tab visible in my screenshot AND you'll see a LabeLPlus pane inside Preferences.

**Comment by neo-neo1** [comment]
@p0psicles where you able to look in this? It would be greatly appreciated. If you have additional questions just ask, I'm more than willing to assist in any way possible to get this done. Unfortunately coding is not my strength.

**Reply by p0psicles** [reply]
If I'm already spending 1,5 hour trying to install the plugin, I'm quickly losing interest. Then the default label plugin isn't that bad for something that works out of the box. 

I'm running on windows. So do I need a native windows client to get this to work?
